graph/builder.go

```diff
@@ -50,8 +312,1497 @@ func (b *Builder) Stop() error {
 	log.Info("Builder shutting down...")
 	defer log.Debug("Builder shutdown complete")
 
+	// Our filtered chain view could've only been started if
+	// AssumeChannelValid isn't present.
+	if !b.cfg.AssumeChannelValid {
+		if err := b.cfg.ChainView.Stop(); err != nil {
+			return err
+		}
+	}
+
 	close(b.quit)
 	b.wg.Wait()
 
 	return nil
 }
+
+// syncGraphWithChain attempts to synchronize the current channel graph with
+// the latest UTXO set state. This process involves pruning from the channel
+// graph any channels which have been closed by spending their funding output
+// since we've been down.
+func (b *Builder) syncGraphWithChain() error {
+	// First, we'll need to check to see if we're already in sync with the
+	// latest state of the UTXO set.
+	bestHash, bestHeight, err := b.cfg.Chain.GetBestBlock()
+	if err != nil {
+		return err
+	}
+	b.bestHeight = uint32(bestHeight)
+
+	pruneHash, pruneHeight, err := b.cfg.Graph.PruneTip()
+	if err != nil {
+		switch {
+		// If the graph has never been pruned, or hasn't fully been
+		// created yet, then we don't treat this as an explicit error.
+		case err == channeldb.ErrGraphNeverPruned:
```

Comment:
I know this commit is a pure code move. But I think if we're touching this code we should make it more robust by using `errors.Is`, especially if we move things from one package to another and are potentially wrapping errors.
Also, feels like the `if` and `switch` could be combined. Fine with me if this happens in a later commit.

Reply:
errors.Is addressed later to keep this commit pure.

Addressed the if/switch suggestion now too :)